<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebuggerConfigWidget</class>
 <widget class="QWidget" name="DebuggerConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>698</width>
    <height>483</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Debugger Configuration</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <property name="whatsThis">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Debugger</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="gdbPath_label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Debugger executable:</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <property name="buddy">
         <cstring>kcfg_gdbPath</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KUrlRequester" name="kcfg_gdbPath">
        <property name="toolTip">
         <string>Gdb executable</string>
        </property>
        <property name="whatsThis">
         <string>To run &quot;gdb&quot; binary from $PATH, leave this field empty. To run custom gdb, for example, for a different architecture, enter the executable name here. You can either run gdb from $PATH, but with a different name (say, &quot;arm-gdb&quot;), by typing the name here, or specify full path to the gdb executable.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="debuggingShell_label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If you want gdb to be executed by a special shell script, which for example properly sets the library path, insert it here e.g.: /path/to/script.sh --scriptArguments&lt;/p&gt;&lt;p&gt;Note: your script'll be invoked like this: /path/to/script.sh --scriptArguments debugger --debuggerArguments. So script.sh &lt;span style=&quot; font-weight:600;&quot;&gt;must&lt;/span&gt; invoke &amp;quot;debugger&amp;quot; executable manually.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
        <property name="text">
         <string>&amp;Shell script:</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <property name="buddy">
         <cstring>kcfg_debuggingShell</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="KUrlRequester" name="kcfg_debuggingShell">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If you want gdb to be executed by a special shell script, which for example properly sets the library path, insert it here e.g.: /path/to/script.sh --scriptArguments&lt;/p&gt;&lt;p&gt;Note: your script'll be invoked like this: /path/to/script.sh --scriptArguments debugger --debuggerArguments. So script.sh &lt;span style=&quot; font-weight:600;&quot;&gt;must&lt;/span&gt; invoke &amp;quot;debugger&amp;quot; executable manually.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Options</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Display static &amp;members:</string>
        </property>
        <property name="buddy">
         <cstring>kcfg_displayStaticMembers</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="kcfg_displayStaticMembers">
        <property name="whatsThis">
         <string>Displaying static members makes GDB slower in
producing data within KDE and Qt.
It may change the &quot;signature&quot; of the data
which QString and friends rely on,
but if you need to debug into these values then
check this option.</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Display &amp;demangled names:</string>
        </property>
        <property name="buddy">
         <cstring>kcfg_asmDemangle</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="kcfg_asmDemangle">
        <property name="whatsThis">
         <string>When displaying the disassembled code you
can select to see the methods' mangled names.
However, non-mangled names are easier to read.</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Start Debugger with:</string>
        </property>
        <property name="buddy">
         <cstring>kcfg_startWith</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="kcfg_startWith">
        <item>
         <property name="text">
          <string>Application Output</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GDB Console</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Frame Stack</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="whatsThis">
      <string>This script is intended for the actual commands needed to connect to a remotely running executable.
	shell sleep 5	wait for remote program to start
	target remote ...	connect to the remote debugger
	continue	[optional] run debugging to the first breakpoint</string>
     </property>
     <property name="title">
      <string>Remote Debugging</string>
     </property>
     <layout class="QGridLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="runShellScript_label">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Script to start remote application&lt;/p&gt;&lt;p&gt;&lt;/p&gt;&lt;p&gt;This shell script is run after the Gdb Config script has been sourced by gdb.&lt;/p&gt;&lt;p&gt;It's purpose is to actually start the remote process.&lt;/p&gt;&lt;p&gt;1) Find a way to execute a command remotely - rsh, ssh, telnet, ...&lt;/p&gt;&lt;p&gt;2a) Execute &amp;quot;gdbserver ... application&amp;quot; on target.&lt;/p&gt;&lt;p&gt;or if your executable contains the gdb stub:&lt;/p&gt;&lt;p&gt;2b) Execute &amp;quot;application&amp;quot; on target.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string comment="Shell script to be executed at run time">R&amp;un shell script:</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <property name="buddy">
         <cstring>kcfg_runShellScript</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="KUrlRequester" name="kcfg_runShellScript">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Script to start remote application&lt;/p&gt;&lt;p&gt;&lt;/p&gt;&lt;p&gt;This shell script is run after the Gdb Config script has been sourced by gdb.&lt;/p&gt;&lt;p&gt;It's purpose is to actually start the remote process.&lt;/p&gt;&lt;p&gt;1) Find a way to execute a command remotely - rsh, ssh, telnet, ...&lt;/p&gt;&lt;p&gt;2a) Execute &amp;quot;gdbserver ... application&amp;quot; on target.&lt;/p&gt;&lt;p&gt;or if your executable contains the gdb stub:&lt;/p&gt;&lt;p&gt;2b) Execute &amp;quot;application&amp;quot; on target.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="configGdbScript_label">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This script is sourced by gdb when the debugging starts.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Gdb &amp;config script:</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <property name="buddy">
         <cstring>kcfg_configGdbScript</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="KUrlRequester" name="kcfg_runGdbScript">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Script to connect with remote application&lt;/p&gt;&lt;p&gt;For example:&lt;/p&gt;&lt;p&gt;	shell sleep 5	 #wait for remote program to start&lt;/p&gt;&lt;p&gt;	target remote ...	#connect to the remote debugger&lt;/p&gt;&lt;p&gt;	continue	#[optional] run debugging to the first breakpoint.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="runGdbScript_label">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Script to connect with remote application&lt;/p&gt;&lt;p&gt;For example:&lt;/p&gt;&lt;p&gt;	shell sleep 5	 #wait for remote program to start&lt;/p&gt;&lt;p&gt;	target remote ...	#connect to the remote debugger&lt;/p&gt;&lt;p&gt;	continue	#[optional] run debugging to the first breakpoint.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string comment="Gdb script to be executed at run time.">Run &amp;gdb script:</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <property name="buddy">
         <cstring>kcfg_runGdbScript</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KUrlRequester" name="kcfg_configGdbScript">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This script is sourced by gdb when the debugging starts.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>kcfg_gdbPath</tabstop>
  <tabstop>kcfg_debuggingShell</tabstop>
  <tabstop>kcfg_displayStaticMembers</tabstop>
  <tabstop>kcfg_asmDemangle</tabstop>
  <tabstop>kcfg_startWith</tabstop>
  <tabstop>kcfg_configGdbScript</tabstop>
  <tabstop>kcfg_runShellScript</tabstop>
  <tabstop>kcfg_runGdbScript</tabstop>
 </tabstops>
 <includes>
  <include location="local">kurlrequester.h</include>
  <include location="local">klineedit.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kurlrequester.h</include>
  <include location="local">klineedit.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kurlrequester.h</include>
  <include location="local">klineedit.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kurlrequester.h</include>
  <include location="local">klineedit.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kurlrequester.h</include>
  <include location="local">klineedit.h</include>
  <include location="local">kpushbutton.h</include>
 </includes>
 <resources/>
 <connections/>
</ui>
